### PR TITLE
refactor epoll event reading into helper

### PR DESCRIPTION
**Pull request by kov**
## Summary
- share code for reading epoll events on eBPF side
- add integration test and helper workload covering epoll syscalls and edge cases

## Testing
- `cargo test -p pinchy-common`
- `cargo test --test integration -- --ignored` *(fails: requires special environment)*

------
https://chatgpt.com/codex/tasks/task_e_68b3616ca9e883319567969d2c70c256

### CODE REVIEW

File: pinchy-ebpf/src/basic_io.rs [hunk 1]
```diff
@@ -264,24 +239,14 @@ pub fn syscall_exit_basic_io(ctx: TracePointContext) -> u32 {
...
             syscalls::SYS_epoll_ctl => {
                 let data = data_mut!(entry, epoll_ctl);
                 data.epfd = args[0] as i32;
                 data.op = args[1] as i32;
                 data.fd = args[2] as i32;
-                let event_ptr = args[3] as *const EpollEvent;
-                if !event_ptr.is_null() {
-                    data.event = unsafe { bpf_probe_read_user(event_ptr).unwrap_or_default() };
-                }
+                read_epoll_events(args[3] as u64, 1, core::slice::from_mut(&mut data.event));
```
[the remark this answers is not in the document]
Comment (kov) [reply pinchy-ebpf/src/basic_io.rs:249]: I think it's fine if the helper checks for it, which it does.
